src/patches.rs

```diff
@@ -3,30 +3,22 @@
 mod atexit;
 #[cfg(feature = "std")]
 mod lstat;
-#[cfg(all(feature = "std", esp_idf_version = "4.3"))]
-mod pthread_rwlock;
 
 #[allow(dead_code)]
 pub struct PatchesRef(
     *mut core::ffi::c_void,
     *mut core::ffi::c_void,
-    *mut core::ffi::c_void,
 );
 
 /// A hack to make sure that the rwlock implementation is linked to the final executable
```

Comment:
The doc comment needs updating too

Reply:
Oops the diff messed up here, but I'm referring to the comment below

Reply:
updated it, though i am not to sure myself what __LSTAT_INTERNAL_REFERENCE is, though the other thing we link is more clear about the atexit ;D

Reply:
so i just call it together "certain symbols" ;D